Models/getModels.js

```diff
@@ -16,13 +17,61 @@ exports.selectCategories = () => {
     });
 };
 
-exports.selectReviews = () => {
-  return db
-    .query(
-      `
-      SELECT *, COUNT(review_id) AS comment_count FROM reviews group by review_id;
-  `,
-    )
+exports.selectReviews = (categories, sortBy, order, query) => {
+  return doesCategoryExist(categories)
+    .then(() => {
+      const sortAllowed = [
+        "owner",
+        "title",
+        "review_id",
+        "category",
+        "review_img_url",
+        "review_body",
+        "created_at",
+        "votes",
+        "designer",
+        "comment_count",
+        "default",
+      ];
+      const orderAllowed = ["asc", "desc"];
+
+      const queries = [];
+      let queryString =
+        "SELECT *, COUNT(review_id) AS comment_count FROM reviews ";
+
+      if (categories) {
+        queryString += "WHERE category = $1 ";
+        queries.push(categories);
+      }
+
+      queryString += "GROUP BY review_id ";
+
+      if (sortBy) {
```

Comment:
This logic is either/or and won't allow you to have a sort and order query at the same time. Try to join them together, and maybe use default values if one or the other isn't provided at all.